templates/Dockerfile.j2

```diff
@@ -41,38 +41,80 @@ RUN set -ex && for esdirs in config data logs; do \
 
 # Install x-pack and also the ingest-{agent,geoip} modules required for Filebeat
 {% if artifacts_dir -%}
-RUN elasticsearch-plugin install --batch "{{ x_pack_url_root }}/x-pack-{{ elastic_version }}.zip" && \
-    for PLUGIN in {{ ingest_plugins }}; do \
+  {% if 'oss' not in image_flavor -%}
+RUN elasticsearch-plugin install --batch "{{ x_pack_url_root }}/x-pack-{{ elastic_version }}.zip"
+  {%- endif %}
+RUN for PLUGIN in {{ ingest_plugins }}; do \
       elasticsearch-plugin install --batch "{{ plugin_url_root }}/$PLUGIN/build/distributions/$PLUGIN-{{ elastic_version }}.zip"; \
     done
 {% elif staging_build_num -%}
+  {% if 'oss' not in image_flavor -%}
 RUN for PLUGIN in x-pack {{ ingest_plugins }}; do \
+  {% else -%}
+RUN for PLUGIN in {{ ingest_plugins }}; do \
+  {%- endif %}
       eval ES_JAVA_OPTS="-Des.plugins.staging={{ staging_build_num }}" elasticsearch-plugin install --batch "$PLUGIN"; \
     done
-{% elif elastic_version.endswith('-SNAPSHOT') -%}
-   COPY {{ base_xpack_path }}/x-pack-{{ elastic_version }}.zip /tmp/x-pack-{{ elastic_version }}.zip
-   RUN elasticsearch-plugin install file:///tmp/x-pack-{{ elastic_version }}.zip
 {% else -%}
+  {% if 'oss' not in image_flavor -%}
 RUN for PLUGIN in x-pack {{ ingest_plugins }}; do \
+  {% else -%}
+RUN for PLUGIN in {{ ingest_plugins }}; do \
+  {%- endif %}
       elasticsearch-plugin install --batch "$PLUGIN"; \
     done
-{% endif-%}
+{%- endif %}
+
+USER root
+
+{# A fair bit of duplication here, but making line breaks and '\' consistent
+   coupled with jinja2 conditionals is complicated.
+-#}
 
+{% if 'oss' in image_flavor -%}
+COPY elasticsearch.yml log4j2.properties config/
+COPY bin/es-docker bin/es-docker
+
+RUN chown elasticsearch:elasticsearch \
+      config/elasticsearch.yml \
+      config/log4j2.properties \
+      bin/es-docker && \
+    chmod 0750 bin/es-docker
+{% elif 'basic' in image_flavor -%}
+COPY elasticsearch.yml log4j2.properties config/
+COPY x-pack/log4j2.properties config/x-pack/
+COPY bin/es-docker bin/es-docker
+
+# Note: basic will be available after 6.0.0-beta1
+RUN echo 'xpack.license.self_generated.type: basic' >>config/elasticsearch.yml
+
+RUN chown elasticsearch:elasticsearch \
+      config/elasticsearch.yml \
+      config/log4j2.properties \
+      config/x-pack/log4j2.properties \
+      bin/es-docker && \
+    chmod 0750 bin/es-docker
+{% elif 'platinum' in image_flavor -%}
 COPY elasticsearch.yml certgen_instances.yml log4j2.properties config/
 COPY x-pack/log4j2.properties config/x-pack/
 COPY bin/es-docker bin/es-docker
 
-USER root
+RUN echo 'xpack.license.self_generated.type: trial' >>config/elasticsearch.yml
+
 RUN chown elasticsearch:elasticsearch \
+      config/certgen_instances.yml \
       config/elasticsearch.yml \
       config/log4j2.properties \
       config/x-pack/log4j2.properties \
       bin/es-docker && \
     chmod 0750 bin/es-docker
+{% endif -%}
 
 USER elasticsearch
 
+{% if image_flavor=='platinum' -%}
```

Comment:
and in 97e034012d5a36b824d93943e43afbd99da36ddc and a9bf1a67069333dc5276e94964f62d7ab975da47.